runtime/src/main/java/org/corfudb/protocols/logprotocol/CheckpointEntry.java

```diff
@@ -0,0 +1,209 @@
+package org.corfudb.protocols.logprotocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import org.corfudb.protocols.wireprotocol.DataType;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.util.serializer.Serializers;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Object & serialization methods for in-stream checkpoint
+ * summarization of SMR object state.
+ */
+@ToString(callSuper = true)
+@NoArgsConstructor
+public class CheckpointEntry extends LogEntry {
+
+    @RequiredArgsConstructor
+    public enum CheckpointEntryType {
+        START(1),           // Mandatory: 1st record in checkpoint
+        CONTINUATION(2),    // Optional: 2nd through (n-1)th record
+        END(3);             // Mandatory: final record checkpoint
+
+        public final int type;
+
+        public byte asByte() {
+            return (byte) type;
+        }
+
+        static final Map<Byte, CheckpointEntryType> typeMap =
+                Arrays.stream(CheckpointEntryType.values())
+                        .collect(Collectors.toMap(CheckpointEntryType::asByte, Function.identity()));
+    };
+
+    /**
+     * Constants for use as keys in our 'dict' map.
+     */
+    public static String START_TIME = "Start time";
```

Comment:
well, the size isn't the main issue, it's that simple mistakes like messing up on the case or using UTF-16 instead of UTF-8 will break things. Change it to a enum which writes a byte, there is no reason for this to be a string